google/cloud/internal/oauth2_minimal_iam_credentials_rest_test.cc

```diff
@@ -262,12 +259,61 @@ TEST(MinimalIamCredentialsRestTest, GenerateAccessTokenSuccess) {
   request.lifetime = lifetime;
   request.scopes.push_back(scope);
   request.delegates.push_back(delegate);
-
   auto access_token = stub.GenerateAccessToken(request);
   EXPECT_THAT(access_token, IsOk());
   EXPECT_THAT(access_token->token, Eq("my_access_token"));
 }
 
+TEST(MinimalIamCredentialsRestTest, GenerateAccessTokenWithUniverseDomain) {
+  std::string universe_domain = "my-ud.net";
+  std::string service_account = "foo@somewhere.com";
+  std::chrono::seconds lifetime(3600);
+  std::string response = R"""({
+    "accessToken": "my_access_token",
+    "expireTime": "2022-10-12T07:20:50.52Z"})""";
+  MockHttpClientFactory mock_client_factory;
+  EXPECT_CALL(mock_client_factory, Call).WillOnce([=](Options const&) {
+    auto client = std::make_unique<MockRestClient>();
+    EXPECT_CALL(*client,
+                Post(_, _, A<std::vector<absl::Span<char const>> const&>()))
+        .WillOnce([response, service_account, universe_domain](
+                      RestContext&, RestRequest const& request,
+                      std::vector<absl::Span<char const>> const&) {
+          auto mock_response = std::make_unique<MockRestResponse>();
+          EXPECT_CALL(*mock_response, StatusCode)
+              .WillRepeatedly(Return(rest_internal::HttpStatusCode::kOk));
+          EXPECT_CALL(std::move(*mock_response), ExtractPayload)
+              .WillOnce([response] {
+                return testing_util::MakeMockHttpPayloadSuccess(response);
+              });
+
+          EXPECT_THAT(
+              request.path(),
+              Eq(absl::StrCat("https://iamcredentials.", universe_domain,
+                              "/v1/projects/-/serviceAccounts/",
+                              service_account, ":generateAccessToken")));
+          return std::unique_ptr<RestResponse>(std::move(mock_response));
+        });
+    return std::unique_ptr<rest_internal::RestClient>(std::move(client));
+  });
+  auto mock_credentials = std::make_shared<MockCredentials>();
+  EXPECT_CALL(*mock_credentials, GetToken).WillOnce([](auto tp) {
+    return AccessToken{"test-token", tp};
+  });
+  EXPECT_CALL(*mock_credentials, universe_domain)
+      .WillOnce([&](Options const&) -> StatusOr<std::string> {
+        return universe_domain;
+      });
```

Comment:
optional nit: (this is more of an FYI for next time, don't feel obligated to push another commit)

We can use `::testing::Return` in simple cases:

```cc
  EXPECT_CALL(*mock_credentials, universe_domain).WillOnce(Return(universe_domain));
```